src/entity/object/FallingBlock.php

```diff
@@ -97,13 +101,29 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 		if($this->closed){
 			return false;
 		}
+		$this->fallTime+=1;
 
 		$hasUpdate = parent::entityBaseTick($tickDiff);
 
 		if(!$this->isFlaggedForDespawn()){
 			$world = $this->getWorld();
 			$pos = $this->location->add(-$this->size->getWidth() / 2, $this->size->getHeight(), -$this->size->getWidth() / 2)->floor();
 
+
+			if($this->getBlock()->getId() == VanillaBlocks::ANVIL()->GetId()) {
+
+				$collidedEntities = $world->getCollidingEntities($this->getBoundingBox());
+				foreach($collidedEntities as $ent)
+					if($ent instanceof Living) {
+						$lastCause = $ent->lastDamageCause;
+						if(!($lastCause instanceof EntityDamageByBlockEvent) || $lastCause->getDamager() !== $this->getBlock()) {
```

Comment:
I'm not very clear what the point of this check is. This is just going to prevent entities from receiving a second damage from falling anvil if their last damage was from an anvil, irrespective of which anvil caused it.

Reply:
Yeah now that you mention it It is unexpected behavior. Its goal was to make sure an anvil only damages player once while falling otherwise it damages every tick. I'll fix this.

Reply:
I retrace my last statement it does damage entity RESPECTIVE to which anvil.

I tested this by dropping multiple anvils on me consecutively each one damaging every time.

Reply:
Checking block identity is not reliable. It depends on the World's block cache, which might have been cleared since the damage occurred.